LSi_Correlator.py

```diff
@@ -0,0 +1,425 @@
+            PvNames.OVERLOADINTERVAL: 400,
+            PvNames.ERRORMSG: "",
+            PvNames.START: 0,
+            PvNames.STOP: 0,
+            PvNames.REPETITIONS: 2,
+            PvNames.CURRENT_REPEAT: 0,
+            PvNames.CORRELATION_FUNCTION: [],
+            PvNames.LAGS: [],
+            PvNames.FILENAME: "",
+            PvNames.FILEPATH: "",
+            PvNames.CONNECTED: self.device.isConnected(),
+            PvNames.RUNNING: False,
+            PvNames.SCATTERING_ANGLE: 2.2,
+            PvNames.SAMPLE_TEMP: 300,
+            PvNames.SOLVENT_VISCOSITY: 1100,
+            PvNames.SOLVENT_REFRACTIVE_INDEX: 1.1,
+            PvNames.LASER_WAVELENGTH: 440,
+            PvNames.SIM: 0,
+            PvNames.DISABLE: 0
+        }
+
+        self.alarm_status = Alarm.NO_ALARM
+        self.alarm_severity = Severity.NO_ALARM
+
+        if os.path.isdir(filepath):
+            self.PVValues[PvNames.FILEPATH] = filepath
+        else:
+            self.update_error_pv_print_and_log("LSiCorrelatorDriver: {} is invalid file path".format(filepath), "MAJOR")
+
+        for pv, preset in self.PVValues.items():
+            # Write defaults to device
+            print("setting {} to {}".format(pv, preset))
+            self.update_pv_value(pv, self.SettingPVs[pv].convert_to_pv(preset))
+
+        self.updatePVs()
+
+    def update_error_pv_print_and_log(self, error: str, severity: str = "INFO", src: str = "LSI") -> None:
+        """
+        Updates the error PV with the provided error message, then prints and logs the error
+
+        Args:
+            error: The error message
+            severity (optional): Gives the severity of the message. Expected serverities are MAJOR, MINOR and INFO.
+            src (optional): Gives the source of the message. Default source is LSI (from this IOC).
+        """
+
+        self.update_pv_value(PvNames.ERRORMSG, error)
+        print_and_log(error, severity, src)
+
+    def set_disconnected_alarms(self, in_alarm: bool):
+        """ Sets disconnected alarms if in_alarm is True """
+        if in_alarm:
+            severity = Severity.INVALID_ALARM
+            status = Alarm.TIMEOUT_ALARM
+        else:
+            severity = Severity.INVALID_ALARM
+            status = Alarm.TIMEOUT_ALARM
+
+        self.alarm_status = status
+        self.alarm_severity = severity
+
+        for pv in self.SettingPVs.keys():
+            self.setParamStatus(pv, status, severity)
+
+    def get_pv_value(self, reason):
+        """
+        Returns the value of a PV held in this driver
+
+        Args:
+            reason (str): The name of the PV to get the value of
+        """
+        if reason.endswith(":SP"):
+            # Return value of base PV, not setpoint
+            reason = get_base_pv(reason)
+
+        if reason in self.SettingPVs:
+            # Need to convert internal state to PV (e.g. enum number)
+            pv_value = self.SettingPVs[reason].convert_to_pv(self.PVValues[reason])
```

Comment:
We're calling the convert here in the read and when we do the write in `update_pv_handler`. Does that mean they're going through the converter twice?